scala-api/src/test/scala/com/cloudera/livy/ScalaClientTest.scala

```diff
@@ -0,0 +1,282 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cloudera.livy
+
+import java.io._
+import java.net.URI
+import java.nio.charset.StandardCharsets._
+import java.util._
+import java.util.jar.JarOutputStream
+import java.util.zip.ZipEntry
+
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.util.Success
+import scala.util.Failure
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import org.apache.spark.SparkFiles
+import org.apache.spark.launcher.SparkLauncher
+import org.scalatest.FunSuite
+import org.scalatest.concurrent.ScalaFutures
+
+import com.cloudera.livy.rsc.RSCConf.Entry._
+
+class ScalaClientTest extends FunSuite with ScalaFutures {
+
+  import com.cloudera.livy.client._
+
+  private val timeout = 40
+  private var client: LivyScalaClient = _
+
+  test("test Job Submission") {
+    configureClient(true)
+    try {
+      pingJob()
+      val jobHandle = client.submit(helloJob())
+      val result = Await.result(jobHandle, 10 second)
+      assert(result === "hello")
+    } catch {
+      case e: Exception => throw e
+    } finally {
+      if (client != null) {
+        client.shutdown()
+        client.stop(true)
+      }
+    }
+  }
+
+  test("test Simple Spark Job") {
+    configureClient(true)
+    try {
+      pingJob()
+      val sFuture = client.submit(simpleSparkJob())
+      val result = Await.result(sFuture, 10 second)
+      assert(result === 5)
+    } catch {
+      case e: Exception => throw e
+    } finally {
+      if (client != null) {
+        client.shutdown()
+        client.stop(true)
+      }
+    }
+  }
+
+  test("test Job Failure") {
+    configureClient(true)
+    try {
+      pingJob()
+      val sFuture = client.submit(throwExceptionJob())
+      Thread.sleep(5000)
+      sFuture onComplete {
+        case Success(t) => {
+          println("Should have thrown an exception")
+          fail()
```

Comment:
`fail("Should have thrown an exception")` and get rid of the println.

BTW you should run `mvn install` or `mvn verify` it will check for a bunch of style violations.